main/main.cpp

```diff
@@ -3846,7 +3847,14 @@ int Main::start() {
 		} else if (E->get() == "--install-android-build-template") {
 			install_android_build_template = true;
 #endif // TOOLS_ENABLED
-		} else if (E->get().length() && E->get()[0] != '-' && positional_arg.is_empty()) {
+		} else if (E->get() == "--scene") {
+			E = E->next();
+			if (E) {
+				game_path = E->get();
+			} else {
+				ERR_FAIL_V_MSG(EXIT_FAILURE, "Missing scene path, aborting.");
+			}
```

Comment:
This is a parameter with an argument to the right, but it's not in the section with the rest of the parameters with an argument to the right, which could be a little confusing given the couple of comments related to this.

Does this need to be up here? If so, that's probably fine, and maybe we could remove those comments. But if it'd work fine in that section, it'd probably be better to be there instead.

Reply:
The other section is in `setup()`, while `game_path` only exists in `start()`. I'm not sure how to pass that properly.

Reply:
I just mean like ~20 lines down, here: https://github.com/godotengine/godot/pull/105302/files#diff-18b7c19417a461e4793bb2157831659d96ede991391e10ee5bc51f19453b7685L3866

Reply:
It seems to check whether next argument exists, before checking the first argument. If you use `--scene` without providing path, you won't get any error.
I think this code is wrong. `--script`, `--main-loop` etc. will just get silently ignored without the second argument and the engine runs normally.

Reply:
Hm, I see. So, the way the parameters under the `// Then parameters that have an argument to the right` section are parsed allows, for example, `godot --script` to run as if `--script` wasn't even there. I guess that's an unrelated bug that needs to be fixed :-)

For your new code to correctly give an error if the argument is missing, it needs to be in the section you've put it in. That makes sense!